contracts/core/GlobalReentrancyLock.sol

```diff
@@ -0,0 +1,166 @@
+pragma solidity 0.8.13;
+
+import {PermissionsV2} from "./PermissionsV2.sol";
+import {IGlobalReentrancyLock} from "./IGlobalReentrancyLock.sol";
+
+/// @notice inpsired by the openzeppelin reentrancy guard smart contracts
+/// data container size has been changed.
+/// @dev allows contracts and addresses with the GLOBAL_LOCKER_ROLE to call
+/// in and lock and unlock this smart contract.
+/// once locked, only the original caller that locked can unlock the contract
+/// without the governor emergency unlock functionality.
+/// Governor can unpause if locked but not unlocked.
+abstract contract GlobalReentrancyLock is IGlobalReentrancyLock, PermissionsV2 {
```

Comment:
Add comment that this contract should only be used from CoreRefV2 `globalReentrancyLock` modifier